src/Deque.mo

```diff
@@ -33,46 +75,158 @@ module {
     }
   };
 
-  /// Insert a new element on the front end of the queue
-  public func pushFront<T>(q : Deque<T>, x : T) : Deque<T> {
-    check(List.push(x, q.0), q.1)
+  /// Extend a deque by inserting a new element on the front end.
+  /// Returns a new deque with `element` in the front followed by the elements of `deque`.
+  ///
+  /// This may involve dynamic splitting of the internally used two lists.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Deque "mo:base/Deque";
+  ///
+  /// Deque.pushFront(Deque.pushFront(Deque.empty<Nat>(), 2), 1) // deque with elements [1, 2]
+  /// ```
+  ///
+  /// Runtime: `O(n)` worst-case, amortized to `O(1)`.
+  ///
+  /// Space: `O(n)` worst-case, amortized to `O(1)`.
+  ///
+  /// `n` denotes the number of elements stored in the deque.
+  public func pushFront<T>(deque : Deque<T>, element : T) : Deque<T> {
+    check(List.push(element, deque.0), deque.1)
   };
 
-  /// Inspect the (optional) first element on the front end of the queue
-  public func peekFront<T>(q : Deque<T>) : ?T {
-    switch q {
+  /// Inspect the optional element on the front end of a deque.
+  /// Returns `null` if `deque` is empty. Otherwise, the front element of `deque`.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Deque "mo:base/Deque";
+  ///
+  /// let deque = Deque.pushFront(Deque.pushFront(Deque.empty<Nat>(), 2), 1);
+  /// Deque.peekFront(deque) // => ?1
+  /// ```
+  ///
+  /// Runtime: `O(1)`.
+  ///
+  /// Space: `O(1)`.
+  ///
+  public func peekFront<T>(deque : Deque<T>) : ?T {
+    switch deque {
       case (?(x, f), r) { ?x };
       case (null, ?(x, r)) { ?x };
       case _ { null }
     }
   };
 
-  /// Remove the first element on the front end of the queue; Returns null when empty.
-  public func popFront<T>(q : Deque<T>) : ?(T, Deque<T>) {
-    switch q {
+  /// Shorten a deque by removing the element on the front end.
+  /// Returns `null` if `deque` is empty. Otherwise, it returns a pair of
+  /// the first element and a new deque that contains all the elements of `deque`,
+  /// however, without the front element.
```

Comment:
```suggestion
  /// Remove an element from the front of the deque.
  /// Returns `null` if `deque` is empty. Otherwise, returns a pair of
  /// the removed element and the remaining elements in a new deque.
```
A rephrasing. Seems clearer to me, but wondering what you think.

Reply:
I agree. It is much clearer. I was unsure how to express the functional nature of the call, that it does not change the structure but returns the new modified deque...